### PR TITLE
[SYCL-MLIR] Add SYCL type attribute to `set_captured` op.

### PR DESCRIPTION
**Pull request by jopperm**

### CODE REVIEW

File: polygeist/tools/cgeist/Test/Verification/sycl/raising/host_raising_captures.cpp [hunk 1]
```diff
@@ -16,8 +16,8 @@ class KernelName;
 // CHECK:         %[[cgf:.*]] = llvm.load %arg0 {{.*}} : !llvm.ptr -> !llvm.ptr
 
 // COM: check that we correctly identified the accessors.
-// CHECK:         sycl.host.set_captured %[[lambda]][0] = %[[accessor_1]] : !llvm.ptr, !llvm.ptr
-// CHECK:         sycl.host.set_captured %[[lambda]][1] = %[[accessor_2]] : !llvm.ptr, !llvm.ptr
+// CHECK:         sycl.host.set_captured %[[lambda]][0] = %[[accessor_1]] : !llvm.ptr, !llvm.ptr (!sycl_accessor_1_21llvm2Evoid_w_gb)
+// CHECK:         sycl.host.set_captured %[[lambda]][1] = %[[accessor_2]] : !llvm.ptr, !llvm.ptr (!sycl_accessor_1_21llvm2Evoid_r_gb)
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Test/Verification/sycl/raising/host_raising_captures.cpp:20]: Note to future us: make `sycl` alias op interface output more readable

**Review by sommerlukas — APPROVED** [review]
Just one nit apart from @victor-eds's comment: There's a spelling mistake in the title.

**Review by whitneywhtsang — APPROVED** [review]
No additional comments from other reviewers.

File: mlir-sycl/test/Dialect/SYCL/host.mlir [hunk 2]
```diff
@@ -80,10 +80,10 @@ func.func @set_nd_range_range_with_offset(%handler: !llvm.ptr, %range: !llvm.ptr
 
 // CHECK-LABEL:   func.func @set_captured(
```
[the remark this answers is not in the document]
Comment (jopperm) [reply mlir-sycl/test/Dialect/SYCL/host.mlir:81]: I've added a check for the variant with a pointer value that is not an accessor. Is that what you meant?

**Review by victor-eds — APPROVED** [review]

**Pull request merged**